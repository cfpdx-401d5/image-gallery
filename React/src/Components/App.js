import React from 'react';

import CreateBunnyForm from './CreateBunnyForm';
import Selector from './Selector';
import View from './View';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      viewStyle: null,
      babyBunBuns: [],
    };
    this.handleClick = this.handleClick.bind(this);
    this.deleteBunny = this.deleteBunny.bind(this);
    this.addBunny = this.addBunny.bind(this);
  }


  handleClick(viewValue) {
    console.log(viewValue);
    this.setState({
      viewStyle: viewValue
    });
  }

  deleteBunny(babyBunIndex, _id){
    fetch(`http://localhost:8000/bunniess/${_id}`, {method: 'DELETE'})
      .then(res => res.json())
      .then(res => {
          if(res.message) {
            //res.message;
            let babyBunBunsFiltered = this.state.babyBunBuns.filter((element, i)=>{
                if (i !== babyBunIndex) {
                    return element
                }
            });
            this.setState({
                babyBunBuns: babyBunBunsFiltered
            })
        }
    })
  }

  addBunny(newBunBun){
    fetch('http://localhost:8000/bunnies', {
      method: 'POST',
      mode: 'cors',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      },
      body: JSON.stringify(newBunBun)
    })
      .then(res => res.json())
      .then(res => {
          delete res.__v;
          let babyBunBunsMultiplied = this.state.babyBunBuns.concat(res);
          this.setState({
              babyBunBuns: babyBunBunsMultiplied
          })
      })
  }

  render() {
    return (
      <div>
        <Selector 
          handleClick={this.handleClick}
        />
        <CreateBunnyForm 
          addBunny={this.addBunny}
        />
        <View 
          deleteBunny={this.deleteBunny} 
          babyBunBuns={this.state.babyBunBuns} 
          viewStyle={this.state.viewStyle}
        />
      </div>
    );
  }

  componentDidMount(){
    fetch('http://localhost:8000/bunnies')
      .then(res => res.json())
      .then(bunbuns => {
        this.setState(
          {babyBunBuns: bunbuns}
        )
      })
  }
}

