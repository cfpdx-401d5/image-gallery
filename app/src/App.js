import React, { Component } from 'react';

import ListDisplay from './displays/ListDisplay';
import ThumbnailDisplay from './displays/ThumbnailDisplay';
import GalleryDisplay from './displays/GalleryDisplay';
import DisplaySelector from './DisplaySelector';
import FormContainer from './image_form/FormContainer';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayType: 'list',
      images: [{
        id: 'id1',
        title: 'Cute Bunny',
        description: 'Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?',
        url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
      },
      {
        id: 'id2',
        title: 'Upside Down Sloth',
        description: 'Sloths just love to hang out.',
        url: 'http://kids.nationalgeographic.com/content/dam/kids/photos/animals/Mammals/Q-Z/sloth-beach-upside-down.jpg.adapt.945.1.jpg'
      },
      {
        id: 'id3',
        title: 'Waving Panda',
        description: 'When a panda waves, you wave back!',
        url: 'http://media4.s-nbcnews.com/j/newscms/2016_36/1685951/ss-160826-twip-05_8cf6d4cb83758449fd400c7c3d71aa1f.nbcnews-ux-2880-1000.jpg'
      }
      ],
    };
    this.setDisplay = this.setDisplay.bind(this);
    this.addImage = this.addImage.bind(this);
    this.deleteImage = this.deleteImage.bind(this);
  }

  setDisplay(whatDisplay) {
    this.setState({ displayType: whatDisplay });
  }

  addImage(e, imageSubmission) {
    e.preventDefault();
    // let newImgArray = [...this.state.images]; // copy the images[] in the state
    // newImgArray.push(imageSubmission); // make new image array

    // this.setState({ images: newImgArray });
    fetch('http://localhost:3001/images', {
      method: 'POST',
      mode: 'cors',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      },
      body: JSON.stringify(imageSubmission)
    })
      .then(res => res.json())
      .then(res => {
        let newImgArray = [...this.state.images];
        delete res.__v;
        newImgArray.push(res);
        this.setState({ images: newImgArray });
      });
  }

  deleteImage(imageID) {
    // let newImgArray = this.state.images.filter(image => {
    //   return (image.id !== imageID);
    // });

    // this.setState({ images: newImgArray });
    fetch(`http://localhost:3001/images/${imageID}`, { method: 'DELETE' })
      .then(res => res.json())
      .then(res => {
        if (res.message) {
          let newImgArray = this.state.images.filter(image => {
            return (image.id !== imageID);
          });
          this.setState({ images: newImgArray });
        }
      });
  }

  render() {
    let DisplayType;

    if (this.state.displayType === 'list') DisplayType = ListDisplay;
    else if (this.state.displayType === 'thumbnail') DisplayType = ThumbnailDisplay;
    else if (this.state.displayType === 'gallery') DisplayType = GalleryDisplay;

    return (
        <div>
          <h1>Image Gallery</h1>
          <FormContainer addImage={this.addImage} />
          <DisplaySelector setDisplay={this.setDisplay} />
          <DisplayType images={this.state.images} deleteImage={this.deleteImage} />
        </div>
    );
  }

  componentDidMount() {
    fetch('http://localhost:3001/images')
      .then(res => res.json())
      .then(resImages => {
        this.setState({ images: resImages });
      });
  }
  
}

