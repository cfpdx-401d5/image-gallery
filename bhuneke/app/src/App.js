import React, { Component } from 'react';
import AddResource from './components/editing/AddResource';
import GalleryDisplay from './components/display/Gallery';
import ThumbnailDisplay from './components/display/Thumbnail';
import DetailDisplay from './components/display/Detail';
import fetcher from './helpers/fetcher';

const GALLERY_TYPE = 'gallery';
const LIST_TYPE = 'list';
const THUMBNAIL_TYPE = 'thumbnail';

export default class App extends Component {

    constructor(props) {
        super(props);
        this.state = {
            selectedView: '',
            resources: []
        };
        this.onViewSelect = this.onViewSelect.bind(this);
        this.onDelete = this.onDelete.bind(this);
        this.onAdd = this.onAdd.bind(this);
    }

    onViewSelect(e, selected) {
        e.preventDefault();
        this.setState({
            selectedView: selected
        });
    }
    doFetch() {
        fetcher({
            path: '/resources',
            method: 'GET',
        })
        .then(res => res.json())
        .then(resources => {
            this.setState({ resources });
        })
        .catch(err => { console.log(err); });
    }

    componentDidMount() {
        this.doFetch();
    }

    onDelete(id) {
        fetcher({
            path: `/resources/${id}`,
            method: 'DELETE'
        });
        this.doFetch();
    }

    onAdd(newResource) {
        let newData = this.state.resources.slice();
        newData.push(newResource);
        this.setState({
            resources: newData
        });
    }

    render() {
        return (
            <div>
                < ViewSelector onViewSelect={this.onViewSelect} />
                < ViewDisplay onDelete={this.onDelete} selectedView={this.state.selectedView} resources={this.state.resources} />
                < AddResource onAdd={this.onAdd} />
            </div>
        );
    }
}

function ViewSelector(props) {
    return (
        <div className='view-selector'>
            <button onClick={(e, selected) => props.onViewSelect(e, LIST_TYPE)}>List View</button>
            <button onClick={(e, selected) => props.onViewSelect(e, THUMBNAIL_TYPE)}>Thumbnail View</button>
            <button onClick={(e, selected) => props.onViewSelect(e, GALLERY_TYPE)}>Gallery View</button>
        </div>);
}

function ViewDisplay(props) {
    let selectedView = props.selectedView;
    if (selectedView) {
        return (
            <ListView
                selectedView={selectedView}
                resources={props.resources}
                onDelete={props.onDelete} 
            />);
    } else {
        return <p>Choose View!</p>;
    }
}

function ListView(props) {
    let DisplayType;
    const selectedView = props.selectedView;
    if (selectedView === LIST_TYPE) DisplayType = DetailDisplay;
    else if (selectedView === GALLERY_TYPE) DisplayType = GalleryDisplay;
    else if (selectedView === THUMBNAIL_TYPE) DisplayType = ThumbnailDisplay;

    const resourceList = props.resources.map(resource => {
        return (
            <DisplayType
                key={resource._id}
                resource={resource}
                onDelete={props.onDelete} 
            />);
    });

    return (
        <div>{resourceList}</div>
    );
}

export {
    ViewSelector,
    ViewDisplay,
    ListView,
};
