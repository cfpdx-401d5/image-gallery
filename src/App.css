.App {
  text-align: center;
}


.App-header {
  background-color: #222;
  color: white;
}

div #root {
  text-align: center;
  width: 90%%;
  margin-left: auto;
  margin-right: auto;
  border: .5em solid #bc640b;
}

.viewer {
height: 80%;
border: .5em solid #000;
}

img {
  max-height: 100%;
  max-width: 100%;
  padding: 1em;
}