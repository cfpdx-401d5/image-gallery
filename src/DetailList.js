import React from 'react';
import DetailDisplay from './DetailDisplay';

export default function DetailList(props) {
    return (
        <div>
            {props.babyBunBuns.map( (b, i) => (
                <DetailDisplay deleteBunny={props.deleteBunny} babyBunIndex={i} babyBunBun={b} key={b.id} />
            ))}
        </div>
    )
}