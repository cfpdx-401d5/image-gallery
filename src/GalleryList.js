import React from 'react';
import GalleryDisplay from './GalleryDisplay';

export default function GalleryList(props) {
    return (
        <div>
            {props.babyBunBuns.map( (b, i) => (
                <GalleryDisplay deleteBunny={props.deleteBunny} babyBunIndex={i} showHandler={props.showHandler} showBun={props.showBun} babyBunBun={b} key={b.id} />
            ))}
        </div>
    )
}

GalleryList.propTypes = {
  deleteBunny: React.PropTypes.func,
  showHandler: React.PropTypes.func,
  showBun: React.PropTypes.boolean,
  babyBunBuns: React.PropTypes.array
};