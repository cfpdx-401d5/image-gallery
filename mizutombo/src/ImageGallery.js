import React, { Component } from 'react';

import images from './images';

import ViewSelectorBar from './ViewSelectorBar';
import DetailList from './DetailList';
import GalleryList from './GalleryList';
import ThumbnailList from './ThumbnailList';
import NewImageForm from './NewImageForm';

// establishes default state
export default class ImageGallery extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentView: 'detail',
      images: images,
    };
    this.clickHandler = this.clickHandler.bind(this);
    this.deleteImage = this.deleteImage.bind(this);
    this.addImage = this.addImage.bind(this);
  }

  clickHandler(newView) {
    this.setState({
      currentView: newView
    });
  }

  doFetch() {
    fetch('http://localhost:3000/bunnies', {
      method: 'GET',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      },
    }).then(r => r.json())
    .then(bunnies => {
      console.log(bunnies);
    });
  }

  addImage(imageToAdd) {
    // let newImages = this.state.images.splice();
    let newImages = [...this.state.images]; // makes copy of images array
    newImages.push(imageToAdd);
    this.setState({
      images: newImages
    });
  }

  deleteImage(imageToDelete) {
    let newImages = this.state.images.filter((image, idx) => {
      if (imageToDelete !== image) {
        return image;
      }
    });
    this.setState({
      images: newImages
    });
  }

  render() {
    let contents;
    
    if (this.state.currentView === 'thumbnail') {
      contents = < ThumbnailList images={this.state.images} deleteImage={this.deleteImage} addImage={this.addImage} />;
    }
    else if (this.state.currentView === 'gallery') {
      contents = < GalleryList images={this.state.images} deleteImage={this.deleteImage} addImage={this.addImage} />;
    }
    else {
      contents = < DetailList images={this.state.images} deleteImage={this.deleteImage} addImage={this.addImage} />;
    }
    return (
      <div>
        <ViewSelectorBar clickHandler={this.clickHandler} />
        <br />
        <NewImageForm addImage={this.addImage} />
        {contents}
      </div>
    );
  }

}
