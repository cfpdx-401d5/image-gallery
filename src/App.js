import React from 'react';
import babyBunBuns from '../data/bunnyimages';
import ThumbnailList from './ThumbnailList';
import DetailList from './DetailList';
import GalleryList from './GalleryList';
import CreateBunnyForm from './NewBunnyForm';


export function Selector(props) {
  return (
    <div>
      <button onClick={() => props.handleClick('thumbnail')}>Thumbnail View</button>
      <button onClick={() => props.handleClick('detail')}>Detail View</button>
      <button onClick={() => props.handleClick('gallery')}>Gallery View</button>
    </div>
  )
}

Selector.propTypes = {
  handleClick: React.PropTypes.func
};

export function View(props) {
  if (props.viewStyle === 'thumbnail') {
    return (
      <ThumbnailList 
        deleteBunny={props.deleteBunny} 
        babyBunBuns={props.babyBunBuns}
      />
    )
  }
  else if (props.viewStyle === 'detail') {
    return (
      <DetailList  
        deleteBunny={props.deleteBunny} 
        babyBunBuns={props.babyBunBuns}
      />
    )
  }
  else if (props.viewStyle === 'gallery') {
    return (
      <GalleryList 
        deleteBunny={props.deleteBunny} 
        showHandler={props.showHandler} 
        showBun={props.showBun} 
        babyBunBuns={props.babyBunBuns}
      />
    )
  }
  else {
    return (
      <p>Click a Button to Get Started</p>
    )
  }
}

View.propTypes = {
  deleteBunny: React.PropTypes.func,
  babyBunBuns: React.PropTypes.array,
  viewStyle: React.PropTypes.string
};

export default class ImageGallery extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      viewStyle: null,
      babyBunBuns: babyBunBuns,
    };
    this.handleClick = this.handleClick.bind(this);
    this.deleteBunny = this.deleteBunny.bind(this);
    this.addBunny = this.addBunny.bind(this);
  }

  handleClick(viewValue) {
    console.log(viewValue);
    this.setState({
      viewStyle: viewValue
    });
  }

  deleteBunny(babyBunIndex){
    let babyBunBunsFiltered = this.state.babyBunBuns.filter((element, i)=>{
      if (i !== babyBunIndex) {
        return element
      }
    });
    this.setState({
      babyBunBuns: babyBunBunsFiltered
    })
  }

  addBunny(newBunBun){
    let babyBunBunsMultiplied = this.state.babyBunBuns.concat(newBunBun);
    this.setState({
      babyBunBuns: babyBunBunsMultiplied
    })
  }

  render() {
    return (
      <div>
        <Selector handleClick={this.handleClick}/>
        <CreateBunnyForm addBunny={this.addBunny}/>
        <View 
          deleteBunny={this.deleteBunny} 
          babyBunBuns={this.state.babyBunBuns} 
          viewStyle={this.state.viewStyle}
        />
      </div>
    );
  }
}

