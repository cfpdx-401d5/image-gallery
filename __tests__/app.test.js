import React from 'react';
import { shallow } from 'enzyme'; //this is a renderer that renders to a virtual DOM
import toJson from 'enzyme-to-json';

import ImageGallery from '../src/App.js';
import { Selector } from '../src/App.js';
import { View } from '../src/App.js';

import ThumbnailList from '../src/ThumbnailList.js';
import DetailList from '../src/DetailList.js';
import GalleryList from '../src/GalleryList.js';

import CreateBunnyForm from '../src/NewBunnyForm.js';


describe('Selector component', () => {
    it('creates buttons', () => {
        const wrapper = shallow(<Selector />);
        const wrapped = toJson(wrapper);
        expect(wrapped).toMatchSnapshot();
    })
})

describe('View Component', () => {
    it('toggles view style', () => {
        const wrapper = shallow(<View />);
        const wrapped = toJson(wrapper);
        expect(wrapped).toMatchSnapshot();
    })
})

describe('Overarching ImageGallery Component', () => {
    it('manages state', () => {
        const wrapper = shallow(<ImageGallery />);
        const wrapped = toJson(wrapper);
        expect(wrapped).toMatchSnapshot();
    })
})

describe('ThumbnailList Component', () => {
    const testBunny = [{
        title: 'Baby Bunny Goes Flop',
        description: 'Adorable bunny flopped over on its side',
        url: 'https://s-media-cache-ak0.pinimg.com/736x/16/4e/8a/164e8aff9e9744e0062c9964d65da48b.jpg'
    }];
    it('creates small titled images of babybunbuns', () => {
        const wrapper = shallow(<ThumbnailList babyBunBuns={testBunny}/>);
        const wrapped = toJson(wrapper);
        expect(wrapped).toMatchSnapshot();
    })
})

describe('DetailList Component', () => {
    const testBunny = [{
        title: 'Baby Bunny Goes Flop',
        description: 'Adorable bunny flopped over on its side',
        url: 'https://s-media-cache-ak0.pinimg.com/736x/16/4e/8a/164e8aff9e9744e0062c9964d65da48b.jpg'
    }];
    it('creates a bunnylist with title, link, and description of picture', () => {
        const wrapper = shallow(<DetailList babyBunBuns={testBunny}/>);
        const wrapped = toJson(wrapper);
        expect(wrapped).toMatchSnapshot();
    })
})

describe('Gallery Component', () => {
    const testBunny = [{
        title: 'Baby Bunny Goes Flop',
        description: 'Adorable bunny flopped over on its side',
        url: 'https://s-media-cache-ak0.pinimg.com/736x/16/4e/8a/164e8aff9e9744e0062c9964d65da48b.jpg'
    }];
    it('creates full-sized, titled images of bunbuns with descriptions', () => {
        const wrapper = shallow(<GalleryList babyBunBuns={testBunny}/>);
        const wrapped = toJson(wrapper);
        expect(wrapped).toMatchSnapshot();
    })
})

describe('CreateBunnyForm Component', () => {
    const testBunny = {
        title: 'Baby Bunny Goes Flop',
        description: 'Adorable bunny flopped over on its side',
        url: 'https://s-media-cache-ak0.pinimg.com/736x/16/4e/8a/164e8aff9e9744e0062c9964d65da48b.jpg'
    };
    it('creates a new bunny', () => {
        const wrapper = shallow(<CreateBunnyForm newBunBun={testBunny} />);
        const wrapped = toJson(wrapper);
        expect(wrapped).toMatchSnapshot();
    })
})
//expect(wrapper.find('li.className')).toHaveLength(3);