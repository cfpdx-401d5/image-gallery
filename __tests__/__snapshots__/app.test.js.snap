exports[`CreateBunnyForm Component creates a new bunny 1`] = `
<form
  onSubmit={[Function]}>
  <label>
    Bunny Title: 
    <input
      onChange={[Function]}
      required={true}
      type="text"
      value="" />
  </label>
  <label>
    Bunny Url: 
    <input
      onChange={[Function]}
      required={true}
      type="text"
      value="" />
  </label>
  <label>
    Bunny Description: 
    <input
      onChange={[Function]}
      required={true}
      type="text"
      value="" />
  </label>
  <input
    type="submit"
    value="Add a Bunny!" />
</form>
`;

exports[`DetailList Component creates a bunnylist with title, link, and description of picture 1`] = `
<div>
  <DetailDisplay
    babyBunBun={
      Object {
        "description": "Adorable bunny flopped over on its side",
        "title": "Baby Bunny Goes Flop",
        "url": "https://s-media-cache-ak0.pinimg.com/736x/16/4e/8a/164e8aff9e9744e0062c9964d65da48b.jpg",
      }
    }
    babyBunIndex={0} />
</div>
`;

exports[`Gallery Component creates full-sized, titled images of bunbuns with descriptions 1`] = `
<div>
  <h4>
    Baby Bunny Goes Flop
  </h4>
  <img
    alt="Baby Bunny Goes Flop"
    height="100px"
    src="https://s-media-cache-ak0.pinimg.com/736x/16/4e/8a/164e8aff9e9744e0062c9964d65da48b.jpg" />
</div>
`;

exports[`Overarching ImageGallery Component manages state 1`] = `
<div>
  <Selector
    handleClick={[Function]} />
  <CreateBunnyForm
    addBunny={[Function]} />
  <View
    babyBunBuns={
      Array [
        Object {
          "description": "Adorable bunny flopped over on its side",
          "id": "1",
          "title": "Baby Bunny Goes Flop",
          "url": "https://s-media-cache-ak0.pinimg.com/736x/16/4e/8a/164e8aff9e9744e0062c9964d65da48b.jpg",
        },
        Object {
          "description": "Baby bunny goes shopping and only gets the essentials",
          "id": "2",
          "title": "Bringin Home the Bacon",
          "url": "http://www.oukongstick.com/wp-content/uploads/2016/09/3-42.jpg",
        },
        Object {
          "description": "Baby bunnies inside of drinking glasses",
          "id": "3",
          "title": "Cruel, or Cute?",
          "url": "https://encrypted-tbn1.gstatic.com/images?q=tbn:ANd9GcQd-m0xBUS65gRgOUrMfk5CQW8b-pvAU6XE6CpDPgy2DqQx39ozEA",
        },
      ]
    }
    deleteBunny={[Function]}
    viewStyle={null} />
</div>
`;

exports[`Selector component creates buttons 1`] = `
<div>
  <button
    onClick={[Function]}>
    Thumbnail View
  </button>
  <button
    onClick={[Function]}>
    Detail View
  </button>
  <button
    onClick={[Function]}>
    Gallery View
  </button>
</div>
`;

exports[`ThumbnailList Component creates small titled images of babybunbuns 1`] = `
<div>
  <h4>
    Baby Bunny Goes Flop
  </h4>
  <img
    alt="Baby Bunny Goes Flop"
    height="100px"
    src="https://s-media-cache-ak0.pinimg.com/736x/16/4e/8a/164e8aff9e9744e0062c9964d65da48b.jpg" />
  <br />
  <button
    onClick={[Function]}>
    Delete Bunny
  </button>
</div>
`;

exports[`View Component toggles view style 1`] = `
<p>
  Click a Button to Get Started
</p>
`;
